app/utils/post.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Posts} from '@mm-redux/constants';
+import {isFromWebhook, isSystemMessage} from '@mm-redux/utils/post_utils';
+import {displayUsername} from '@mm-redux/utils/user_utils';
+
+import type {Post} from '@mm-redux/types/posts';
+import type {UserProfile} from '@mm-redux/types/users';
+
+export function areConsecutivePost(post: Post, previousPost: Post) {
+    let consecutive = false;
+
+    if (post && previousPost) {
+        const postFromWebhook = Boolean(post?.props?.from_webhook); // eslint-disable-line camelcase
+        const prevPostFromWebhook = Boolean(previousPost?.props?.from_webhook); // eslint-disable-line camelcase
+        if (previousPost && previousPost.user_id === post.user_id &&
+            post.create_at - previousPost.create_at <= Posts.POST_COLLAPSE_TIMEOUT &&
```

Comment:
Although this block of conditions is valid, it can be broken down into small constants.  The reading experience will benefit from it.